Framework/DataHandling/src/SaveNXcanSAS.cpp

```diff
@@ -382,7 +382,7 @@ void addData1D(H5::Group &data, Mantid::API::MatrixWorkspace_sptr workspace) {
 
   //-----------------------------------------
   // Add Q with units  + uncertainty definition
-  const auto qValue = workspace->readX(0);
+  const auto &qValue = workspace->x(0);
```

Comment:
Use `workspace->points(0)` and remove the branching and call to `convertToBinCEntre` below?